# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PBrowser.ipdl
# Commit: e1fbefe38944
# Full Hash: e1fbefe38944eb82d3c2e87a0307b121daf0e830
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-04 04:33:36
# Regressor Bug: 1510569
# File Overlap Count: 1
# Description:
#   Bug 1510569 - Port onProgressChange notifications inside WebProgressChild.jsm to C++ r=baku
#   
#   We do not need to handle onProgressChange64 notifications since the TabChild's
#   web progress events are filtered through an nsBrowserStatusFilter, which
#   truncates onProgresChange64 event values to 32-bit integers and then calls
# ==============================================================================

diff -r 102881458323 -r e1fbefe38944 dom/ipc/PBrowser.ipdl
--- a/dom/ipc/PBrowser.ipdl	Wed Apr 03 17:32:21 2019 +0000
+++ b/dom/ipc/PBrowser.ipdl	Wed Apr 03 17:32:41 2019 +0000
@@ -537,6 +537,11 @@
     async RegisterProtocolHandler(nsString scheme, nsIURI handlerURI, nsString title,
                                   nsIURI documentURI);
 
+    async OnProgressChange(WebProgressData? aWebProgressData,
+                           RequestData aRequestData, int32_t aCurSelfProgress,
+                           int32_t aMaxSelfProgress, int32_t aCurTotalProgress,
+                           int32_t aMaxTotalProgress);
+
     async OnStatusChange(WebProgressData? aWebProgressData,
                          RequestData aRequestData, nsresult aStatus,
                          nsString aMessage);